Failing operator test - KeycloakDeploymentTest.testPodNamePropagation

### Description

In several job runs, inluding https://github.com/keycloak/keycloak/actions/runs/10594540860/job/29358667097, KeycloakDeploymentTest.testPodNamePropagation is failing due to an underlying exception of

Caused by: jakarta.enterprise.inject.UnsatisfiedResolutionException: No bean found for required type [class io.quarkus.opentelemetry.runtime.tracing.DelayedAttributes] and qualifiers [[@jakarta.enterprise.inject.Any()]]

This is not consistently occurring nor has it been reproducable locally. Locally I consistenly observe TracerProducer providing the DelayedAttributes regardless of where other breakpoints are set in attempts to alter the startup.

cc @geoand @mabartos

Quarkus 3.13.3 is used.
The stack trace:
```

2024-08-28T10:29:28.5140978Z jakarta.enterprise.inject.CreationException: Error creating synthetic bean [Le6zQbzkojAYO_OiKIQWJf4lGa4]: jakarta.enterprise.inject.UnsatisfiedResolutionException: No bean found for required type [class io.quarkus.opentelemetry.runtime.tracing.DelayedAttributes] and qualifiers [[@jakarta.enterprise.inject.Any()]]
2024-08-28T10:29:28.5144299Z 	at io.opentelemetry.api.OpenTelemetry_Le6zQbzkojAYO_OiKIQWJf4lGa4_Synthetic_Bean.doCreate(Unknown Source)
2024-08-28T10:29:28.5145539Z 	at io.opentelemetry.api.OpenTelemetry_Le6zQbzkojAYO_OiKIQWJf4lGa4_Synthetic_Bean.create(Unknown Source)
2024-08-28T10:29:28.5146472Z 	at io.opentelemetry.api.OpenTelemetry_Le6zQbzkojAYO_OiKIQWJf4lGa4_Synthetic_Bean.create(Unknown Source)
2024-08-28T10:29:28.5147378Z 	at io.quarkus.arc.impl.AbstractSharedContext.createInstanceHandle(AbstractSharedContext.java:119)
2024-08-28T10:29:28.5148220Z 	at io.quarkus.arc.impl.AbstractSharedContext$1.get(AbstractSharedContext.java:38)
2024-08-28T10:29:28.5148940Z 	at io.quarkus.arc.impl.AbstractSharedContext$1.get(AbstractSharedContext.java:35)
2024-08-28T10:29:28.5149549Z 	at io.quarkus.arc.impl.LazyValue.get(LazyValue.java:32)
2024-08-28T10:29:28.5150412Z 	at io.quarkus.arc.impl.ComputingCache.computeIfAbsent(ComputingCache.java:69)
2024-08-28T10:29:28.5151298Z 	at io.quarkus.arc.impl.ComputingCacheContextInstances.computeIfAbsent(ComputingCacheContextInstances.java:19)
2024-08-28T10:29:28.5152191Z 	at io.quarkus.arc.impl.AbstractSharedContext.get(AbstractSharedContext.java:35)
2024-08-28T10:29:28.5153211Z 	at io.opentelemetry.api.OpenTelemetry_Le6zQbzkojAYO_OiKIQWJf4lGa4_Synthetic_Bean.get(Unknown Source)
2024-08-28T10:29:28.5154201Z 	at io.opentelemetry.api.OpenTelemetry_Le6zQbzkojAYO_OiKIQWJf4lGa4_Synthetic_Bean.get(Unknown Source)
2024-08-28T10:29:28.5155119Z 	at io.quarkus.arc.impl.InstanceImpl.getBeanInstance(InstanceImpl.java:325)
2024-08-28T10:29:28.5155843Z 	at io.quarkus.arc.impl.InstanceImpl.getInternal(InstanceImpl.java:309)
2024-08-28T10:29:28.5156513Z 	at io.quarkus.arc.impl.InstanceImpl.get(InstanceImpl.java:190)
2024-08-28T10:29:28.5157275Z 	at io.quarkus.arc.runtime.BeanContainerImpl.beanInstance(BeanContainerImpl.java:26)
2024-08-28T10:29:28.5158477Z 	at io.quarkus.opentelemetry.runtime.tracing.intrumentation.InstrumentationRecorder.setupVertxTracer(InstrumentationRecorder.java:47)
2024-08-28T10:29:28.5159990Z 	at io.quarkus.deployment.steps.OpenTelemetryProcessor$setupVertx126580776.deploy_0(Unknown Source)
2024-08-28T10:29:28.5160875Z 	at io.quarkus.deployment.steps.OpenTelemetryProcessor$setupVertx126580776.deploy(Unknown Source)
2024-08-28T10:29:28.5161699Z 	at io.quarkus.runner.ApplicationImpl.doStart(Unknown Source)
2024-08-28T10:29:28.5162253Z 	at io.quarkus.runtime.Application.start(Application.java:101)
2024-08-28T10:29:28.5162955Z 	at io.quarkus.runtime.ApplicationLifecycleManager.run(ApplicationLifecycleManager.java:119)
2024-08-28T10:29:28.5163599Z 	at io.quarkus.runtime.Quarkus.run(Quarkus.java:71)
2024-08-28T10:29:28.5164154Z 	at org.keycloak.quarkus.runtime.KeycloakMain.start(KeycloakMain.java:141)
2024-08-28T10:29:28.5164959Z 	at org.keycloak.quarkus.runtime.cli.command.AbstractStartCommand.run(AbstractStartCommand.java:42)
2024-08-28T10:29:28.5165713Z 	at picocli.CommandLine.executeUserObject(CommandLine.java:2030)
2024-08-28T10:29:28.5166513Z 	at picocli.CommandLine.access$1500(CommandLine.java:148)
2024-08-28T10:29:28.5167230Z 	at picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent(CommandLine.java:2465)
2024-08-28T10:29:28.5167959Z 	at picocli.CommandLine$RunLast.handle(CommandLine.java:2457)
2024-08-28T10:29:28.5168472Z 	at picocli.CommandLine$RunLast.handle(CommandLine.java:2419)
2024-08-28T10:29:28.5169077Z 	at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:2277)
2024-08-28T10:29:28.5169696Z 	at picocli.CommandLine$RunLast.execute(CommandLine.java:2421)
2024-08-28T10:29:28.5170186Z 	at picocli.CommandLine.execute(CommandLine.java:2174)
2024-08-28T10:29:28.5170750Z 	at org.keycloak.quarkus.runtime.cli.Picocli.parseAndRun(Picocli.java:138)
2024-08-28T10:29:28.5171422Z 	at org.keycloak.quarkus.runtime.KeycloakMain.main(KeycloakMain.java:101)
2024-08-28T10:29:28.5172237Z 	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
2024-08-28T10:29:28.5173001Z 	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
2024-08-28T10:29:28.5173647Z 	at io.quarkus.bootstrap.runner.QuarkusEntryPoint.doRun(QuarkusEntryPoint.java:62)
2024-08-28T10:29:28.5174557Z 	at io.quarkus.bootstrap.runner.QuarkusEntryPoint.main(QuarkusEntryPoint.java:33)
2024-08-28T10:29:28.5175950Z Caused by: jakarta.enterprise.inject.UnsatisfiedResolutionException: No bean found for required type [class io.quarkus.opentelemetry.runtime.tracing.DelayedAttributes] and qualifiers [[@jakarta.enterprise.inject.Any()]]
2024-08-28T10:29:28.5177216Z 	at io.quarkus.arc.impl.InstanceImpl.bean(InstanceImpl.java:288)
2024-08-28T10:29:28.5177816Z 	at io.quarkus.arc.impl.InstanceImpl.getInternal(InstanceImpl.java:309)
2024-08-28T10:29:28.5178605Z 	at io.quarkus.arc.impl.InstanceImpl.get(InstanceImpl.java:190)
2024-08-28T10:29:28.5179772Z 	at io.quarkus.opentelemetry.runtime.AutoConfiguredOpenTelemetrySdkBuilderCustomizer$TracingResourceCustomizer$1.apply(AutoConfiguredOpenTelemetrySdkBuilderCustomizer.java:71)
2024-08-28T10:29:28.5181502Z 	at io.quarkus.opentelemetry.runtime.AutoConfiguredOpenTelemetrySdkBuilderCustomizer$TracingResourceCustomizer$1.apply(AutoConfiguredOpenTelemetrySdkBuilderCustomizer.java:65)
2024-08-28T10:29:28.5183093Z 	at io.opentelemetry.sdk.autoconfigure.AutoConfiguredOpenTelemetrySdkBuilder.lambda$mergeCustomizer$13(AutoConfiguredOpenTelemetrySdkBuilder.java:648)
2024-08-28T10:29:28.5184349Z 	at io.opentelemetry.sdk.autoconfigure.ResourceConfiguration.configureResource(ResourceConfiguration.java:111)
2024-08-28T10:29:28.5185537Z 	at io.opentelemetry.sdk.autoconfigure.AutoConfiguredOpenTelemetrySdkBuilder.build(AutoConfiguredOpenTelemetrySdkBuilder.java:439)
2024-08-28T10:29:28.5186619Z 	at io.quarkus.opentelemetry.runtime.OpenTelemetryRecorder$1.apply(OpenTelemetryRecorder.java:81)
2024-08-28T10:29:28.5187491Z 	at io.quarkus.opentelemetry.runtime.OpenTelemetryRecorder$1.apply(OpenTelemetryRecorder.java:54)
2024-08-28T10:29:28.5188394Z 	at io.opentelemetry.api.OpenTelemetry_Le6zQbzkojAYO_OiKIQWJf4lGa4_Synthetic_Bean.createSynthetic(Unknown Source)
2024-08-28T10:29:28.5190127Z 	... 39 more
```

--- a/operator/src/test/java/org/keycloak/operator/testsuite/integration/KeycloakDeploymentTest.java
+++ b/operator/src/test/java/org/keycloak/operator/testsuite/integration/KeycloakDeploymentTest.java
@@ -35,7 +35,6 @@ import io.quarkus.test.junit.QuarkusTest;
 
 import org.awaitility.Awaitility;
 import org.junit.jupiter.api.Assumptions;
-import org.junit.jupiter.api.Disabled;
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.condition.EnabledIfSystemProperty;
 import org.keycloak.operator.Config;
@@ -47,7 +46,6 @@ import org.keycloak.operator.crds.v2alpha1.deployment.Keycloak;
 import org.keycloak.operator.crds.v2alpha1.deployment.KeycloakStatusCondition;
 import org.keycloak.operator.crds.v2alpha1.deployment.ValueOrSecret;
 import org.keycloak.operator.crds.v2alpha1.deployment.spec.BootstrapAdminSpec;
-import org.keycloak.operator.crds.v2alpha1.deployment.spec.FeatureSpec;
 import org.keycloak.operator.crds.v2alpha1.deployment.spec.HostnameSpecBuilder;
 import org.keycloak.operator.crds.v2alpha1.deployment.spec.TracingSpecBuilder;
 import org.keycloak.operator.testsuite.unit.WatchedResourcesTest;
@@ -74,8 +72,6 @@ import static java.util.concurrent.TimeUnit.SECONDS;
 import static org.assertj.core.api.Assertions.assertThat;
 import static org.junit.jupiter.api.Assertions.assertEquals;
 import static org.junit.jupiter.api.Assertions.assertTrue;
-import static org.keycloak.operator.controllers.KeycloakDeploymentDependentResource.KC_TRACING_RESOURCE_ATTRIBUTES;
-import static org.keycloak.operator.controllers.KeycloakDeploymentDependentResource.KC_TRACING_SERVICE_NAME;
 import static org.keycloak.operator.testsuite.utils.CRAssert.assertKeycloakStatusCondition;
 import static org.keycloak.operator.testsuite.utils.K8sUtils.deployKeycloak;
 import static org.keycloak.operator.testsuite.utils.K8sUtils.getResourceFromFile;
@@ -388,53 +384,6 @@ public class KeycloakDeploymentTest extends BaseOperatorTest {
         deployKeycloak(k8sclient, kc, true);
 
         CRAssert.assertKeycloakAccessibleViaService(k8sclient, kc, false, httpPort);
-    }
-
-    @Test
-    @Disabled
-    public void testPodNamePropagation() {
-        var kc = getTestKeycloakDeployment(true);
-        var featureSpec = new FeatureSpec();
-        featureSpec.setEnabledFeatures(List.of("opentelemetry"));
-        kc.getSpec().setFeatureSpec(featureSpec);
-        kc.getSpec().getAdditionalOptions().add(new ValueOrSecret("tracing-enabled", "true"));
-        kc.getSpec().getAdditionalOptions().add(new ValueOrSecret("log-level", "io.opentelemetry:fine"));
-        deployKeycloak(k8sclient, kc, true);
-
-        Awaitility.await()
-                .ignoreExceptions()
-                .untilAsserted(() ->
-                        assertThat(k8sclient
-                                .pods()
-                                .inNamespace(namespace)
-                                .withLabel("app", "keycloak")
-                                .list()
-                                .getItems()
-                                .size()).isNotZero());
-
-        var pods = k8sclient
-                .pods()
-                .inNamespace(namespace)
-                .withLabels(Constants.DEFAULT_LABELS)
-                .list()
-                .getItems();
-
-        var envVars = pods.get(0).getSpec().getContainers().get(0).getEnv();
-        assertThat(envVars).isNotNull();
-        assertThat(envVars).isNotEmpty();
-
-        var serviceNameEnv = envVars.stream().filter(f -> f.getName().equals(KC_TRACING_SERVICE_NAME)).findAny().orElse(null);
-        assertThat(serviceNameEnv).isNotNull();
-        assertThat(serviceNameEnv.getValue()).isEqualTo(kc.getMetadata().getName());
-
-        var resourceAttributesEnv = envVars.stream().filter(f -> f.getName().equals(KC_TRACING_RESOURCE_ATTRIBUTES)).findAny().orElse(null);
-        assertThat(resourceAttributesEnv).isNotNull();
-
-        var expectedAttributes = Map.of(
-                "k8s.namespace.name", kc.getMetadata().getNamespace()
-        ).entrySet().stream().map(e -> e.getKey() + "=" + e.getValue()).collect(Collectors.joining(","));
-
-        assertThat(resourceAttributesEnv.getValue()).isEqualTo(expectedAttributes);
     }
 
     @Test

--- a/operator/src/test/java/org/keycloak/operator/testsuite/integration/TracingDeploymentTest.java
+++ b/operator/src/test/java/org/keycloak/operator/testsuite/integration/TracingDeploymentTest.java
@@ -1,0 +1,84 @@
+/*
+ * Copyright 2024 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.keycloak.operator.testsuite.integration;
+
+import io.quarkus.test.junit.QuarkusTest;
+import org.awaitility.Awaitility;
+import org.junit.jupiter.api.Test;
+import org.keycloak.operator.Constants;
+import org.keycloak.operator.crds.v2alpha1.deployment.ValueOrSecret;
+import org.keycloak.operator.crds.v2alpha1.deployment.spec.TracingSpec;
+
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.keycloak.operator.controllers.KeycloakDeploymentDependentResource.KC_TRACING_RESOURCE_ATTRIBUTES;
+import static org.keycloak.operator.controllers.KeycloakDeploymentDependentResource.KC_TRACING_SERVICE_NAME;
+import static org.keycloak.operator.testsuite.utils.K8sUtils.deployKeycloak;
+
+@QuarkusTest
+public class TracingDeploymentTest extends BaseOperatorTest {
+
+    @Test
+    public void podNamePropagation() {
+        var kc = getTestKeycloakDeployment(false);
+        kc.getSpec().setStartOptimized(false);
+
+        var tracingSpec = new TracingSpec();
+        tracingSpec.setEnabled(true);
+        kc.getSpec().setTracingSpec(tracingSpec);
+
+        kc.getSpec().getAdditionalOptions().add(new ValueOrSecret("log-level", "io.opentelemetry:fine"));
+        deployKeycloak(k8sclient, kc, true);
+
+        Awaitility.await()
+                .ignoreExceptions()
+                .untilAsserted(() ->
+                        assertThat(k8sclient
+                                .pods()
+                                .inNamespace(namespace)
+                                .withLabel("app", "keycloak")
+                                .list()
+                                .getItems()
+                                .size()).isNotZero());
+
+        var pods = k8sclient
+                .pods()
+                .inNamespace(namespace)
+                .withLabels(Constants.DEFAULT_LABELS)
+                .list()
+                .getItems();
+
+        var envVars = pods.get(0).getSpec().getContainers().get(0).getEnv();
+        assertThat(envVars).isNotNull();
+        assertThat(envVars).isNotEmpty();
+
+        var serviceNameEnv = envVars.stream().filter(f -> f.getName().equals(KC_TRACING_SERVICE_NAME)).findAny().orElse(null);
+        assertThat(serviceNameEnv).isNotNull();
+        assertThat(serviceNameEnv.getValue()).isEqualTo(kc.getMetadata().getName());
+
+        var resourceAttributesEnv = envVars.stream().filter(f -> f.getName().equals(KC_TRACING_RESOURCE_ATTRIBUTES)).findAny().orElse(null);
+        assertThat(resourceAttributesEnv).isNotNull();
+
+        var expectedAttributes = Map.of(
+                "k8s.namespace.name", kc.getMetadata().getNamespace()
+        ).entrySet().stream().map(e -> e.getKey() + "=" + e.getValue()).collect(Collectors.joining(","));
+
+        assertThat(resourceAttributesEnv.getValue()).isEqualTo(expectedAttributes);
+    }
+}

